iam-login-service/src/test/java/it/infn/mw/iam/test/notification/CustomTemplateNotificationTests.java

```diff
@@ -0,0 +1,154 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package it.infn.mw.iam.test.notification;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import it.infn.mw.iam.IamLoginService;
+import it.infn.mw.iam.notification.NotificationProperties;
+import it.infn.mw.iam.persistence.model.IamEmailNotification;
+import it.infn.mw.iam.registration.PersistentUUIDTokenGenerator;
+import it.infn.mw.iam.registration.RegistrationRequestDto;
+import it.infn.mw.iam.test.core.CoreControllerTestSupport;
+import it.infn.mw.iam.test.util.WithAnonymousUser;
+import it.infn.mw.iam.test.util.notification.MockNotificationDelivery;
+import it.infn.mw.iam.test.util.oauth.MockOAuth2Filter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.context.WebApplicationContext;
+
+import static it.infn.mw.iam.test.util.AuthenticationUtils.adminAuthentication;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertThat;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.authentication;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.log;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = {IamLoginService.class, NotificationTestConfig.class,
+    CoreControllerTestSupport.class})
+@WebAppConfiguration
+@Transactional
+@WithAnonymousUser
+@TestPropertySource(properties = {
+        "notification.disable=false",
+        "notification.useCustomTemplates=true",
+        "notification.customTemplateLocation=C://Users//fou43474//iam//iam-login-service//src//test//resources//templates-on-disk/"
```

Comment:
This won't likely work outside of your machine.

Reply:
Yes that's a good point. Unfortunately anything in @TestPropertySource has to be static, otherwise I'd get the working dir at runtime. Do you have any suggestions based on your experience with Spring Boot what we could do here instead please?